pkg/init/backend/interactive.go

```diff
@@ -100,19 +101,23 @@ func (o *InteractiveBackend) SelectStarterProject(devfile parser.DevfileObj, fla
 	return &starterProjects[starter], nil
 }
 
-func (o *InteractiveBackend) PersonalizeName(devfile parser.DevfileObj, flags map[string]string) error {
+func (o *InteractiveBackend) PersonalizeName(devfile parser.DevfileObj, flags map[string]string) (parser.DevfileObj, error) {
 	name, err := o.askerClient.AskName(fmt.Sprintf("my-%s-app", devfile.Data.GetMetadata().Name))
 	if err != nil {
-		return err
+		return parser.DevfileObj{}, err
 	}
-	return devfile.SetMetadataName(name)
+	metadata := devfile.Data.GetMetadata()
+	metadata.Name = name
+	devfile.Data.SetMetadata(metadata)
+	return devfile, nil
 }
 
-func (o *InteractiveBackend) PersonalizeDevfileconfig(devfileobj parser.DevfileObj) error {
+func (o *InteractiveBackend) PersonalizeDevfileConfig(devfileobj parser.DevfileObj) (parser.DevfileObj, error) {
 	// TODO: Add tests
 	config, err := getPortsAndEnvVar(devfileobj)
+	var dFile parser.DevfileObj
```

Comment:
You could name it `zeroDevfile` to indicate it is the zero-value (we want to pass when an error occurs)